src/dotty/tools/dotc/transform/IsInstanceOfEvaluator.scala

```diff
@@ -0,0 +1,148 @@
+package dotty.tools.dotc
+package transform
+
+import dotty.tools.dotc.util.Positions._
+import TreeTransforms.{MiniPhaseTransform, TransformerInfo}
+import core._
+import Contexts.Context, Types._, Constants._, Decorators._, Symbols._
+import TypeUtils._, TypeErasure._
+
+
+/** Implements partial `isInstanceOf` evaluation according to the matrix on:
+ *  https://github.com/lampepfl/dotty/issues/1255
```

Comment:
Instead of referring to an issue, I'd appreciate it if the comment for this phase explained in more details what it did and why, as it is I have trouble figuring out what's going on here :).